neurons/validators/scraper_validator.py

```diff
@@ -454,11 +537,50 @@ async def query_and_score(self, strategy=QUERY_MINERS.RANDOM):
 
             dataset = QuestionsDataset()
             tools = random.choice(self.tools)
+            # Get current time in UTC
+            current_time = datetime.now(pytz.utc)
+            
+            # Track which UIDs already have recent organic results
+            uids_with_recent_results = []
+            recent_organic_results = []
+            recent_organic_rewards = []
+            
+            # Check organic history for each available UID
+            for uid in self.neuron.available_uids:
+                # Get hotkey for this UID
+                neuron = next((n for n in self.neuron.metagraph.neurons if n.uid == uid), None)
+                if not neuron:
+                    continue
+                    
+                hotkey = neuron.hotkey
+                
+                # Check if this hotkey has organic history
+                if hotkey in self.neuron.organic_query_state.organic_history:
+                    for synapse, is_failed, rewards in self.neuron.organic_query_state.organic_history[hotkey]:
+                        # Only consider if tools match
+                        if set(synapse.tools) == set(tools):
+                            # Parse the synapse start date
+                            synapse_time = datetime.strptime(
+                                synapse.start_date, 
+                                "%Y-%m-%dT%H:%M:%SZ"
+                            ).replace(tzinfo=pytz.utc)
+                            
+                            # Check if within last 30 minutes
+                            time_diff_minutes = (current_time - synapse_time).total_seconds() / 60
+                            if time_diff_minutes <= 30 and not is_failed:
+                                uids_with_recent_results.append(uid)
+                                recent_organic_results.append(synapse)
+                                recent_organic_rewards.append(rewards)
+                                break
```

Comment:
maybe move as function for readability

Reply:
I created get_uids_with_recent_organic_results function in ScraperValidator Class